portal-impl/src/com/liferay/portlet/HeaderResponseImpl.java

```diff
@@ -214,8 +214,7 @@ public void setUseDefaultTemplate(Boolean useDefaultTemplate) {
 	}
 
 	private void _addMarkupToHead(
-			String name, String scope, String version, String markup)
-		throws IllegalArgumentException {
```

Comment:
@shuyangzhou I have to put the `IllegalArgumentException` back because otherwise it fails a TCK test:
```
V3HeaderPortletTests_SPEC15_HeaderResponse_addDependency12 results:
Test Failed

Details: Method addDependency(String name, String scope, String version, String markup) - Throws IllegalArgumentException - if markup does not contain valid tags for the document HEAD section. Failed because no exception is raised.
```

Reply:
@ngriffin7a In the code you sent, addDependency() does not even call _addMarkupToHead(). So you are asking you will add that later? If so, that is fine, but don't declare "throws IllegalArgumentException" at the method signature, IllegalArgumentException is an RuntimeException, which is an unchecked exception, we don't declare those.